initial autofill infra

sashei opened the pull request:
resolves #609

jhugman commented on docs/updating_libraries.md:
```diff
@@ -1,17 +1,26 @@
 # External Dependencies
 
+With the addition of the Credential Provider API in iOS 12, you will need to use **Xcode 10** (beta 5) and **Swift 4.2** (`swiftlang-1000.0.32.1 clang-1000.10.39`) to work on Lockbox.
```
Chore: we need to update this to new Xcode when the GM comes out.

jhugman commented on Shared/Common/Resources/BaseConstants.swift:
```diff
@@ -0,0 +1,45 @@
+    }
+}
+
+enum SharedUserDefaultKey: String {
```
Does this really need to be called `Shared`?

jhugman commented on Shared/Store/BaseAccountStore.swift:
```diff
@@ -0,0 +1,64 @@
+
+class BaseAccountStore {
+    internal var localKeychainWrapper: KeychainWrapper
+    internal var sharedKeychainWrapper: KeychainWrapper
```
I don't understand the need for two keychains here.

I'd also *strongly* advise you to convince yourself that you are using the keychain as the one used in the `lockbox-fxa-sync-ios` library. 

This might be as simple as using the `KeychainWrapper.sharedAppContainerKeychain` everywhere we're using `KeychainWrapper.standard`.

jhugman commented on lockbox-ios/Common/Extensions/UserDefaults+.swift:
```diff
@@ -1,34 +1,23 @@
 
     var onRecordUsageData: Observable<Bool> {
-        return self.on(setting: .recordUsageData, type: Bool.self)
+        return self.on(setting: LocalUserDefaultKey.recordUsageData.rawValue, type: Bool.self)
```
Again, I'm not convinced of the value of having extension and app version of `UserDefaults`.

jhugman commented on lockbox-ios/Common/Resources/Constants.swift:
```diff
@@ -6,19 +6,30 @@
+    static let clientID = "98adfa37698f255b"
+}
+
+extension Constant.setting {
```
Cute.

sashei commented:
I'm addressing these PR comments, but I would like to merge the PR with upgrading infrastructure as well so that we can support users in migrating their set preferences to the shared `UserDefaults` and `Keychain` instances.
